internal/transport/controlbuf.go

```diff
@@ -846,7 +856,8 @@ func (l *loopyWriter) handle(i interface{}) error {
 	case *outFlowControlSizeRequest:
 		l.outFlowControlSizeRequestHandler(i)
 	case closeConnection:
-		return l.closeConnectionHandler()
+		l.closeConnection()
```

Comment:
The only current usage of the `closeConnection` type is from the server-side keepalive code (when the grace period expires). Why do we need this separate type? Why can't we instead simply close the transport and let is call `Close()` on the underlying connection. Is this intended to be way to close the connection *after* completing all pending tasks in the controlbuf, while closing the transport will immediately close the underlying connection without completing pending tasks?

Reply:
Yes, we need to flush any pending writes that may have made it legal to close the transport at this time (vs waiting longer for streams to finish).  This was added recently: #5821